### ISSUE TITLE
The `never` option of `function-whitespace-after` writes a calculation the CSS grammar cannot read

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#257]
Found while fixing [#230](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/230), which stops the same corruption where the parenthesis groups an expression rather than closing a call. Where it really does close a call, the `never` option still writes a sum the CSS grammar cannot read.

### What minimal example or steps are needed to reproduce the bug?

A call standing on one side of an operator inside a calculation:

```css
a { b: calc(var(--x) + 1px); }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/function-whitespace-after": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

Nothing reported. The whitespace behind that parenthesis is not the whitespace this rule is about: it is the space CSS Values 4 requires on both sides of a `+` or a `-` inside a calculation, and the rule has no business taking it away.

### What actually happened?

`1:22 Unexpected whitespace after ")"`, and the fix writes:

```css
a { b: calc(var(--x)+ 1px); }
```

A browser drops that declaration. `+` and `-` inside a math function are only operators when whitespace stands on both sides, so `--x)+ 1px` is a syntax error rather than a sum, and the same goes for `calc(attr(data-x)+ 1px)` and for the `min()`, `max()` and `clamp()` family.

`always` is safe in the other direction only by accident: it asks for whitespace, and whitespace is what the grammar wants there. It writes a space where the operator abuts the parenthesis, `calc(var(--x)-1px)` becoming `calc(var(--x) -1px)`, which is not the sum the author wrote either — one space is not two, and `-1px` after a space is a signed number rather than an operator and a value. So both options can leave a calculation the browser will not read; `never` is the one that does it to code that was correct to begin with.

### What is the cause?

`ACCEPTABLE_AFTER_CLOSING_PAREN` in [`function-whitespace-after/index.js`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/function-whitespace-after/index.js) lists the characters the `always` option may leave un-spaced, and there is no list at all for `never`: it reports whitespace behind any closing parenthesis of a call. Nothing asks what the call stands inside.

### Do you have a proposal to fix the bug?

Ask what the parenthesis stands inside, as [#230](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/230) taught the rule to ask what it closes. `findFunctionArgumentSpans` gives every span the name it was opened by, so a call standing inside `calc`, `min`, `max`, `clamp` and the rest of the math functions can be told from one standing in an ordinary value, and the whitespace beside a `+` or a `-` there left alone under both options.

A narrower reading would do as well: leave the whitespace where the character behind it is a `+` or a `-` with whitespace on the far side too, since that pair is a sum wherever it stands. It reads no names, and it costs nothing to the values this rule is really about.
